xarray/core/variable.py

```diff
@@ -2494,10 +2497,29 @@ def rolling_window(
         )
 
     def coarsen(
-        self, windows, func, boundary="exact", side="left", keep_attrs=None, **kwargs
+        self,
+        windows: Mapping[Any, int],
+        func: str | Callable,
+        boundary: CoarsenBoundaryOptions = "exact",
+        side: SideOptions | Mapping[Any, SideOptions] = "left",
+        keep_attrs=None,
+        **kwargs,
     ):
         """
         Apply reduction function.
+
+        Parameters
+        ----------
+        windows : mapping of hashable to int
+            A mapping from the name of the dimension to create the coarsened block along (e.g. `time`) to the size of
+            the coarsened block.
+        func : function or str name of function
+            Function to use to reduce the values of one block down along one or more axes.
```

Comment:
Maybe mention the signature of the function.
I myself have already run into the problem if not knowing what exactly you have to supply.

Also, I never know from which "pool" you can choose the str names, maybe this can be documented better? (Same for the other docstrings)

Reply:
> Also, I never know from which "pool" you can choose the str names, maybe this can be documented better? (Same for the other docstrings)

Yeah this seems like an issue. It's currently literally just any function in the internal `duck_array_ops` module, which (a) is hacky AF, (b) is not documented or even very easy to document, and (c) could fail (there are fn's in `duck_array_ops` that aren't reductions, for example. I'm not aware of anywhere else in xarray's codebase where we just implicitly allow attempting to get any function from `duck_array_ops` like this.

Reply:
Maybe we could add a public variable in the duck_arr_ops module that collects valid reduction names and add a reference to it?